src/Mailer/Mailer.php

```diff
@@ -0,0 +1,222 @@
+<?php
+/**
+ * CakePHP(tm) : Rapid Development Framework (http://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ * @since         3.1.0
+ * @license       http://www.opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\Mailer;
+
+use ArrayAccess;
+use Cake\Datasource\ModelAwareTrait;
+use Cake\Event\EventListenerInterface;
+use Cake\Mailer\Exception\MissingActionException;
+use Cake\Utility\Inflector;
+
+abstract class Mailer implements ArrayAccess, EventListenerInterface
+{
+    use ModelAwareTrait;
+
+    /**
+     * Layout.
+     *
+     * @var string
+     */
+    public $layout;
+
+    /**
+     * Email instance.
+     *
+     * @var \Cake\Mailer\Email
+     */
+    protected $_email;
+
+    /**
+     * Constructor.
+     *
+     * @param \Cake\Mailer\Email|null $email Email instance.
+     */
+    public function __construct(Email $email = null)
+    {
+        if ($email === null) {
+            $email = new Email();
+        }
+
+        if ($this->layout === null) {
+            $this->layout = Inflector::underscore($this->getName());
+        }
+
+        $this->_email = $email->profile((array)$this);
```

Comment:
Since `send()` always does `$this->_email->profile((array)$this)` doing it constructor seems redundant.